---
title: Конструктор <name> не может вызывать сам себя
ms.date: 07/20/2015
f1_keywords:
- bc30298
- vbc30298
helpviewer_keywords:
- BC30298
ms.assetid: 2d77b7f4-0640-4f89-9c65-f101fd2847c0
ms.openlocfilehash: ef20f74055a07071ef9634973c6852ac58c3143c
ms.sourcegitcommit: bce0586f0cccaae6d6cbd625d5a7b824d1d3de4b
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 04/02/2019
ms.locfileid: "58824724"
---
# <a name="constructor-name-cannot-call-itself"></a>Конструктор "\<имя >" не может вызвать сам себя
Объект `Sub New` процедура в классе или структуре вызывает саму себя.  
  
 Конструктор предназначен для инициализации нового экземпляра класса или структуры при первом создании. Класс или структура может иметь несколько конструкторов, если все они имеют разные списки параметров. Конструктор может вызвать другой конструктор для выполнения его функциональных возможностей в дополнение к свой собственный. Но не имеет смысла для конструктора вызвать сам себя, и на самом деле это приведет к бесконечной рекурсии Если это разрешено.  
  
 **Идентификатор ошибки:** BC30298  
  
## <a name="to-correct-this-error"></a>Исправление ошибки  
  
1.  Проверьте список параметров конструктора. Она должна отличаться от вызова конструктора.  
  
2.  Если вам не требуется вызывать другой конструктор, удалите `Sub New` вызов полностью.  
  
## <a name="see-also"></a>См. также

- [Время существования: Способ создания и уничтожения объектов](../../../visual-basic/programming-guide/language-features/objects-and-classes/object-lifetime-how-objects-are-created-and-destroyed.md)
